sentry/src/main/java/io/sentry/ILogger.java

```diff
@@ -30,4 +32,14 @@
    * @param args the formatting arguments
    */
   void log(SentryLevel level, Throwable throwable, String message, Object... args);
+
+  /**
+   * Whether this logger is enabled for the specified SentryLevel.
+   *
+   * @param level The SentryLevel to test against.
+   * @return True if a log message would be recorded for the level. Otherwise false.
+   */
+  default boolean isEnabled(final @Nullable SentryLevel level) {
+    return true;
```

Comment:
I'm afraid of this though. So we will say log is enabled and incur the cost of whatever the caller was calling this to check for.

Meaning: `isEnabled(debug) { do expensive stuff`;

What about we default to `false`? Error on the side of caution?
To be honest both are suboptional

Reply:
This is just for the sake of raising `isEnabled` method visibility level and not implementing this method in all loggers. `DiagnosticLogger` overwrites it with proper implementation. 

If we default it to `false` we should change `SystemOutLogger` implementation to not log anything or overwrite this method to return always `true`.

Reply:
> If we default it to false we should change SystemOutLogger implementation to not log anything or overwrite this method to return always true.

that sounds good to me

Reply:
I have left it without default implementation so that it does not silently stop working in case some of our users implemented custom loggers.